src/Structure/Content/Menu.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Abstract class for a custom menu.
+ *
+ * @author  Chrispian Burks <chrispian.burks@webdevstudios.com>
+ * @package WebDevStudios\OopsWP\Structure\Content
+ */
+
+namespace WebDevStudios\OopsWP\Structure\Content;
+
+/**
+ * Class Menu
+ *
+ * @package WebDevStudios\OopsWP\Structure\Content
+ * @since   2019-05-01
+ */
+abstract class Menu implements MenuInterface {
+	/**
+	 * Menu Location.
+	 *
+	 * @var string
+	 * @since 2019-05-01
+	 */
+	protected $location;
+
+	/**
+	 * Menu Description.
+	 *
+	 * @var string
+	 * @since 2019-05-01
+	 */
+	protected $description;
+
+	/**
+	 * Callback to register the menu with WordPress.
+	 *
+	 * @author Chrispian H. Burks <chrispian.burks@webdevstudios.com>
+	 * @throws \Exception If validation fails throw an exception.
+	 * @since 2019-05-01
+	 */
+	public function register() {
+		$this->validate_location();
+		$this->validate_description();
+
+		register_nav_menu( $this->location, $this->description );
+	}
+
+	/**
+	 * Validate the provided menu location.
+	 *
+	 * @author Chrispian Burks <chrispian.burks@webdevstudios.com>
+	 * @throws \Exception If the menu location is not valid.
+	 * @since  2019-05-01
+	 */
+	private function validate_location() {
+		if ( empty( $this->location ) ) {
```

Comment:
I think we can cut this exception, as it's covered by the second exception. We really only care that the location is a string - it's null by default, so that check would would pass and the exception would be thrown.